File: occasum-core/src/main/java/pw/stamina/occasum/properties/enums/StandardEnumParsingService.java
package pw.stamina.occasum.properties.enums;

import pw.stamina.occasum.properties.PropertyParseException;

final class StandardEnumParsingService<T extends Enum<T>> extends AbstractEnumParsingService<T> {

    StandardEnumParsingService(Class<T> enumClass) {
        super(enumClass);
    }

    @Override
    public T parse(String input) throws PropertyParseException {
        try {
            return Enum.valueOf(enumClass, input);
        } catch (NullPointerException | IllegalArgumentException e) {
            throw new PropertyParseException(e, input);
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/enums/StandardEnumParsingService.kt
package pw.stamina.occasum.properties.enums

import pw.stamina.occasum.properties.PropertyParseException

internal class StandardEnumParsingService<T : Enum<T>>(enumClass: Class<T>) : AbstractEnumParsingService<T>(enumClass) {

    @Throws(PropertyParseException::class)
    override fun parse(input: String): T {
        try {
            return java.lang.Enum.valueOf<T>(enumClass, input)
        } catch (e: IllegalArgumentException) {
            throw PropertyParseException(e, input)
        }
    }
}
